### PR TITLE
IWantToRunWhenConfigurationIsComplete obsoleted

### PR DESCRIPTION
**Pull request by danielmarbach**
Mentioned that we need to write doco in the housekeeping repo

### CODE REVIEW

**Comment by danielmarbach** [comment]
@johnsimons Please review

File: src/NServiceBus.Core/Configure_Obsolete.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+﻿namespace NServiceBus
+{
+    using NServiceBus.Config;
+
+    public partial class Configure
+    {
+        [ObsoleteEx(Message = "Remove when IWantToRunWhenConfigurationIsComplete is obsoleted.", RemoveInVersion = "7.0")]
+        void RegisterWantToRunWhenConfigurationIsCompleteAsInstancePerCall()
```
Comment (SimonCropp) [inline src/NServiceBus.Core/Configure_Obsolete.cs]: why mark privates as obsolete?

Comment (danielmarbach) [inline src/NServiceBus.Core/Configure_Obsolete.cs]: I introduced them as a reminder to remove in next version. Or should I
remove the code and no longer execute them?

Comment (SimonCropp) [inline src/NServiceBus.Core/Configure_Obsolete.cs]: well you have obsoleted IWantToRunWhenConfigurationIsComplete  in 7. as per https://github.com/Fody/Obsolete#steptype it means "Treat as error" will be 6.

So once we go 6 no one can use IWantToRunWhenConfigurationIsComplete   in v6. 

so if no one can use IWantToRunWhenConfigurationIsComplete   then there is no point registering them or running them.

Comment (danielmarbach) [inline src/NServiceBus.Core/Configure_Obsolete.cs]: I realized that when I wrote my question to you. Updating…

**Pull request merged**